### PR TITLE
feat(service-portal): Filter petition routes by scope 

### PR DESCRIPTION
**Pull request by alexandradilja**
Access controlled menu routes in service portal.

Admin Petition view should only be visible and accessible for authenticated users with `@island.is/endorsements:admin` scope. 
Using the userInfo parameter and filtering like noted in the [documentation](https://docs.devland.is/libs/service-portal/core#service-portal-shell)  and is done [here](https://github.com/island-is/island.is/pull/4389). 

At this time the userInfo is returning all scopes listed under service-portal and not the scopes that the user has. There is a task for this here in [Notion]( https://www.notion.so/IDS-returns-protected-scopes-in-callback-query-and-token-response-4a6410326aca4181a825ff73f3af4256). When this has been fixed Admin view in menu will become visible to admin users only. Until then the `@island.is/endorsements:admin` is listed on all users and the admin view will continue to be visible like it currently is, and therefor the Petition module will be kept under a feature flag until userInfo.scopes returns the correct scopes. 

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/service-portal/endorsements/src/index.ts [hunk 1]
```diff
@@ -17,29 +20,64 @@ export const endorsementsModule: ServicePortalModule = {
   ],
 }
 
+const hasGeneralPetitionAdminScope = (user: ServicePortalModuleProps) => {
+  for (const [key, value] of Object.entries(user.userInfo.scopes)) {
+    console.log(value)
```
Comment (jGeir) [inline libs/service-portal/endorsements/src/index.ts:25]: Is this supposed to be here?

File: libs/service-portal/endorsements/src/index.ts [hunk 2]
```diff
@@ -17,29 +20,64 @@ export const endorsementsModule: ServicePortalModule = {
...
+      },
+      {
+        name: m.endorsementsAdmin,
+        path: ServicePortalPath.PetitionListAdmin,
+        enabled: userInfo.scopes.includes(EndorsementsScope.admin),
+        render: () => lazy(() => import('./screens/ViewPetitionAdmin')),
+      },
+    ]
+
+    console.log(userInfo)
```
Comment (jGeir) [inline libs/service-portal/endorsements/src/index.ts:61]: Is this console log supposed to be here?

File: libs/service-portal/endorsements/src/index.ts [hunk 3]
```diff
@@ -17,29 +20,64 @@ export const endorsementsModule: ServicePortalModule = {
...
+      //     render: () => lazy(() => import('./screens/PetitionsAdmin')),
+      //   },
+      //   {
+      //     name: m.endorsementsAdmin,
+      //     path: ServicePortalPath.PetitionListAdmin,
+      //     enabled: userInfo.scopes.includes(EndorsementsScope.admin),
+      //     render: () => lazy(() => import('./screens/ViewPetitionAdmin')),
+      //   },
+      // )
+    // }
```
Comment (jGeir) [inline libs/service-portal/endorsements/src/index.ts:79]: Is this commented code required for future development?

**Review by jGeir — APPROVED** [review]
LGTM - Minor comments.

**Review by albinagu — APPROVED** [review]

**Pull request merged**